src/js/components/Diagram/doc.js

```diff
@@ -15,9 +15,35 @@ export const doc = Diagram => {
     .intrinsicElement('svg');
 
   DocumentedDiagram.propTypes = {
+    animation: PropTypes.oneOfType([
+      PropTypes.bool,
+      PropTypes.oneOf(['pulse', 'draw']),
+      PropTypes.shape({
+        type: PropTypes.oneOf(['pulse', 'draw']),
+        delay: PropTypes.number,
+        duration: PropTypes.number,
+        size: PropTypes.oneOf(['xsmall', 'small', 'medium', 'large', 'xlarge']),
+      }),
+    ]).description('Animation to be used by entire Diagram'),
     connections: PropTypes.arrayOf(
       PropTypes.shape({
         anchor: PropTypes.oneOf(['center', 'vertical', 'horizontal']),
+        animation: PropTypes.oneOfType([
+          PropTypes.bool,
+          PropTypes.oneOf(['pulse', 'draw']),
+          PropTypes.shape({
+            type: PropTypes.oneOf(['pulse', 'draw']),
+            delay: PropTypes.number,
+            duration: PropTypes.number,
+            size: PropTypes.oneOf([
+              'xsmall',
+              'small',
+              'medium',
+              'large',
+              'xlarge',
+            ]),
+          }),
```

Comment:
This code doesn't have to be dulicated.

Reply:
Recent commits now solve this. Now they're under a variable called `ANIMATION_PROP` that get plugged in at both the connection level and component level